qdrant-landing/content/articles/new-recommendation-api.md

```diff
@@ -0,0 +1,336 @@
+We can use different approaches to determine the path of traversing the HNSW graph by changing how we calculate the score of a 
+candidate point during traversal. The default behaviour is based on pure distance, but Qdrant 1.6 exposes two strategies for the 
+recommendation API.
+
+### Average vector
+
+The default strategy, called `average_vector` is the previous one, based on the average of positive and negative examples. It 
+simplifies the recommendations process and converts it into a single vector search. It supports both point IDs and vectors as 
+parameters. For example, you can get recommendations based on past interactions with existing points combined with query vector 
+embedding. Internally, that mechanism is based on the averages of positive and negative examples and was calculated with the 
+following formula:
+
+```
+average_vector = avg(positive_vectors) + ( avg(positive_vectors) - avg(negative_vectors) )
+```
+
+The `average_vector` converts the problem of recommendations into a single vector search.
+
+### The new hotness - Best score
+
+The new strategy is called `best_score`. It does not rely on averages and is more flexible. It allows you to pass just negative 
+samples and uses a slightly more sophisticated algorithm under the hood.
+
+The best score is chosen at every step of HNSW graph traversal. We separately calculate the distance between a traversed point 
+and every positive and negative example. In the case of the best score strategy, **there is no single query vector anymore, but a 
+bunch of positive and negative queries**. As a result, for each sample in the query, we have a set of distances, one for each 
+sample. In the next step, we simply take the best scores for positives and negatives, creating two separate values. Best scores 
+are just the closest distances of a query to positives and negatives. The idea is: **if a point is closer to any negative than to 
+any positive example, we do not want it**. We penalize being close to the negatives, so instead of using the similarity value 
+directly, we check if it’s closer to positives or negatives. The following formula is used to calculate the score of a traversed 
+potential point:
+
+```rust
+if best_positive_score > best_negative_score {
+    score = best_positive_score
+} else {
+    score = -(best_negative_score * best_negative_score)
+}
+```
+
+If the point is closer to the negatives, we penalize it by taking the negative squared value of the best negative score. For a 
+closer negative, the score of the candidate point will always be lower or equal to zero, making the chances of choosing that point 
+significantly lower. However, if the best negative score is higher than the best positive score, we still prefer those that are 
+further away from the negatives. That procedure effectively **pulls the traversal procedure away from the negative examples**.
```

Comment:
Clarifying why the square of the negative score is used (as opposed to other forms of penalization) could add a little more colour and context perhaps?

Reply:
I guess that's just one of the options available, and I don't want to discuss the details that much.